src/screenlock/screenlock.cpp

```diff
@@ -204,11 +209,36 @@ void ScreenLock::setScreenLocked(bool value)
 {
     // TODO Make the view a lock screen view (title? stacking layer?)
     if (m_lockscreenVisible != value) {
+        if (m_allowSkippingLockScreen) {
+            // Allow disabling lock screen, but only if the device is not locked
+            if (value && !m_deviceIsLocked) {
+                return;
```

Comment:
It is not immediately obvious to me where from / when this function is called. But if it is done also as an reaction to mce sending tklock change signals over dbus, simply ignoring the change leads to mce and lipstick getting out of sync regarding tklock status -> weird problems -> lipstick should make an effort to bounce also mce side back to m_lockscreenVisible == false like state [e.g. unlockScreen() call? Which still seems to use legacy dbus interface btw]